src/clean-core/to_string.hh

```diff
@@ -1,27 +1,106 @@
 #pragma once
 
-#include <clean-core/string.hh>
+#include <clean-core/fwd.hh>
 
 namespace cc
 {
-inline string to_string(char value) { return string::filled(1, value); }
-inline string to_string(bool value) { return value ? "true" : "false"; }
-inline string to_string(char const* value) { return value == nullptr ? "[nullptr]" : value; }
-inline string to_string(string_view value) { return value; }
-inline string to_string(nullptr_t) { return "nullptr"; }
+string to_string(char value);
+string to_string(bool value);
+string to_string(char const* value);
+string to_string(string_view value);
+string to_string(nullptr_t);
```

Comment:
Im not sure whether not including string here is worth not being able to inline these calls. Technically all but the first one could just return `char const*` anyway

Reply:
I've discussed this with julius and the voted for this solution because calling to_string in a hot loop is not a good idea anways.
I don't like `char const*` return value because that would be wholly inconsistent (why is `to_string(2) + " foo"` working but not `to_string(true) + " foo"`? sounds like a big headache).
btw, `string_view` is not null-terminated so `char const*` would not work.

Reply:
I see, i was just thinking since the format internals rely so heavily on these functions and currently live in a different TU. Completely agree on the consistency part.

Reply:
Aren't all `cc` internals currently in a single TU?

Reply:
Right!